dask/dataframe/io.py

```diff
@@ -708,13 +710,25 @@ def one_path_one_key(path, key, start, stop, columns, chunksize, lock):
                           stop, empty, chunksize))
         name = 'read-hdf-' + token
 
-        dsk = dict(((name, i), (_pd_read_hdf, path, key, lock,
-                                 {'start': s,
-                                  'stop': s + chunksize,
-                                  'columns': empty.columns}))
-                    for i, s in enumerate(range(start, stop, chunksize)))
-
-        divisions = [None] * (len(dsk) + 1)
+        if not division_column:
+            dsk = dict(((name, i), (_pd_read_hdf, path, key, lock,
+                {'start': s,
+                 'stop': s + chunksize,
+                 'columns': empty.columns}))
+                       for i, s in enumerate(range(start, stop, chunksize)))
+            divisions = [None] * (len(dsk) + 1)
+        else:
+            buffer = _pd_read_hdf(path, key, lock, {'chunksize': chunksize,
+                                                    'start': start,
+                                                    'stop': stop})
```

Comment:
Not per the documentaiton of `read_hdf` (http://pandas.pydata.org/pandas-docs/version/0.17.0/generated/pandas.read_hdf.html) this will return an iterator. Only 2 chunks (at most) should be loaded into RAM at a time.